lib/src/db.rs

```diff
@@ -484,10 +484,17 @@ impl Storelike for Db {
         if include_external {
             prefix = "".as_bytes();
         }
+        let pool = ThreadPool::new(self.njobs());
         let resource_iter = self.resources.scan_prefix(prefix);
-        let resources = resource_iter
-            .map(|item| process_resource(item))
-            .collect::<Vec<Resource>>();
+        let (send, recv) = channel::bounded(0);
+        for item in resource_iter {
+            let (send, item)= (send.clone(),item.clone());
+            pool.execute(move || {
+                send.send(process_resource(item));
+                });
+            }
+            drop(send);
+        let resources=recv.try_iter().collect::<Vec<Resource>>();
```

Comment:
Remove`collect::<Vec<Resource>>()` and it will be iterator. I am actually not sure about performance improvement, validating it now - cargo criterion and cargo bench give different results. This one can be closed - I will create clean PR in a few minutes.